### PR TITLE
adds custom arguments for authorization_code

### PR DESCRIPTION
**Pull request by remcojansen**
This adds support for --custom arguments to the authorization_code flow, which enables users to provide additional arguments to the authorization request as needed.

### CODE REVIEW

File: oidc_config.go [hunk 1]
```diff
@@ -36,3 +36,14 @@ func (c *Config) DiscoverEndpoints() {
 	assignIfEmpty(&c.IntrospectionEndpoint, discoveryJson["introspection_endpoint"].(string))
 	assignIfEmpty(&c.UserinfoEndpoint, discoveryJson["userinfo_endpoint"].(string))
 }
+
+type CustomArgs []string
+
+func (c *CustomArgs) String() string {
```
Comment (jentz) [inline oidc_config.go:42]: Is the idea these will be used later on?

Comment (remcojansen) [reply oidc_config.go:42]: I figured it might be useful for other commands too?

File: oidc.go [hunk 2]
```diff
@@ -99,6 +99,11 @@ func HandleOpenIDFlow(clientID, clientSecret, scopes, callbackURL, discoveryEndp
 	query.Set("scope", scopes)
 	query.Set("redirect_uri", callbackURL)
 
+	for _, arg := range customArgs {
+		kv := strings.SplitN(arg, "=", 2)
+		query.Set(kv[0], kv[1])
```
Comment (jentz) [inline oidc.go:104]: We are missing input validation in more places but I guess we might want to parse the CLI inout and save the results in a map which would push the validation up.

File: oidc_config.go (hunk 1 above)
Comment (remcojansen) [reply oidc_config.go:42]: Oh you meant the Strings() method I guess. I got the impression it was required by the flags package, but didn't look further into it.

Comment (jentz) [reply oidc_config.go:42]: I meant that this String() function is not used anywhere. Is it required by flags?

Comment (jentz) [reply oidc_config.go:42]: Let's not worry about that for now, we can adjust validation and stuff later.

**Review by jentz — APPROVED** [review]

**Pull request merged**